compiler/src/formatting/format.re

```diff
@@ -1132,6 +1132,52 @@ and resugar_pattern_list_inner = (patterns: list(Parsetree.pattern)) => {
   };
 }
 
+and process_list_comments = items => {
+  // We have to compose this list by hand because of the complexity of if a list item
+  // is followed by a comment, the comma must come before the comment.
+  // It also impacts how we force a new line for a line ending comment at the end of a list
+  // without introducing an extra blank line when bringing the indentation back in again
```

Comment:
On my list of things to do is create a new parsetree node for lists, now that it's a part of the compiler. Seems like I should hurry up and do that so we can rebase this on top to prevent this from getting even more complex?

Reply:
Sure, if you'd like; I don't think it should be too much work to rebase this off of such a feature